src/operator/operator_common.h

```diff
@@ -414,6 +401,14 @@ void FCompExFallback(const nnvm::NodeAttrs& attrs,
           << ") == " << param << ".shape[0] (" << rsp.shape()[0] << ").";          \
   }
 
+template<typename xpu, int ndim, typename DType>
+inline mshadow::Tensor<xpu, ndim, DType> AllocateTempDataForSparseHandling(
+  const OpContext& op_ctx, const mshadow::Shape<ndim>& shape) {
+  Resource rsc = ResourceManager::Get()->Request(op_ctx.run_ctx.ctx,
```

Comment:
I'm not actually sure of the internals here and how it might deadlock. I can use Alloc/Free instead.